runtime/src/server/middleware/get_page_handler.ts

```diff
@@ -106,38 +106,42 @@ export function get_page_handler(
 			},
 			fetch: (url: string, opts?: any) => {
 				const parsed = new URL.URL(url, `http://127.0.0.1:${process.env.PORT}${req.baseUrl ? req.baseUrl + '/' :''}`);
+					
+				opts = opts ? Object.assign({}, opts) : {};
 
-				if (opts) {
-					opts = Object.assign({}, opts);
-
-					const include_cookies = (
+				const includeCredentials = (
+					opts.credentials !== 'omit' && (
 						opts.credentials === 'include' ||
-						opts.credentials === 'same-origin' && parsed.origin === `http://127.0.0.1:${process.env.PORT}`
-					);
+						parsed.origin === `http://127.0.0.1:${process.env.PORT}`
```

Comment:
it might be nice to encapsulate this in a method like `isSameOrigin` so that in the future we can more easily make the URL to fetch from configurable. It also would help better explain the logic here

Reply:
Actually, isn't this a no-op?  `parsed` was hardcoded to this above, right? So you could probably simplify by just making this check be `opts.credentials !== 'omit'`

Reply:
No, `parsed` is constructed from `url`, relative to `http://127.0.0.1/...` if it did not contain a domain.